### PR TITLE
[feature] add host processor

### PR DESCRIPTION
**Pull request by mahone3297**

### CODE REVIEW

**Comment by HonzaMac** [comment]
What is the difference from #1026 ?

**Comment by mahone3297** [comment]
@HonzaMac I don't have a look at the PRs before implmenting it. I think it's the same. 😭

**Comment by HonzaMac** [comment]
Your is caching `gethostname()` which is ok, but still does not handle all return types from that function

**Comment by mahone3297** [comment]
@HonzaMac >but still does not handle all return types from that function
What's the meaning? I can't understand, can you explain?

**Comment by HonzaMac** [comment]
It could [return FALSE ](http://php.net/manual/en/function.gethostname.php)
So you will have string or boolean(false) in extra arguments. 
i.e. elastic search will not like that and omit this message at all.

**Comment by mahone3297** [comment]
@HonzaMac 
* function __invoke will return array
* $record['extra']['host'] will have default value ''
so, I don't think return bool(false) is a good option.

**Comment by HonzaMac** [comment]
The default value will be immediately overwritten by value from the constructor and that could be string or false.

I.e. in `$record['extra']['host'] = 'localhost';` or `$record['extra']['host'] = false;`

**Comment by staabm** [comment]
IMO those very simple processor classes should be defined in-line and dont need to be bundled with monolog. re-using them is not always possible (e.g. long running app, vs. web-sapi based app etc).

I would propose adding a example to the docs like

```php
$hostProcessor = new class {
    private static $host;
    
    public function __construct()
    {
        self::$host = gethostname();
    }
    
    public function __invoke(array $record): array
    {
        $record['extra']['host'] = self::$host;
 
        return $record;
    }
};
```

**Comment by mahone3297** [comment]
@HonzaMac 
* host field, I think it's a string tyle, so I think `$record['extra']['host'] = ‘’;` is better than  `$record['extra']['host'] = false;`
* `$record['extra']['host'] = 'localhost';`
 maybe 'localhost' is not the correct value, so i think `''` is better than `'localhost'`, at least we can know get host failed

@staabm I don't think so. you can say it's very simple, but, I think many people will re-use it, so put it in monolog will be better, many people can be benefit from it.
ps: you say defined in-line, I am writing the code below in my project, it's more simple
```
$logger->pushProcessor(function (array $record) {
    $record['extra']['host'] = $this->host;
    return $record;
});
```

**Comment by staabm** [comment]
> ps: you say defined in-line, I am writing the code below in my project, it's more simple

event better!

**Comment by HonzaMac** [comment]
@mahone3297 
But your code does not do what you are saying ;)
add something like this: 
```
$hostname = gethostname();
if ($hostname === FALSE) {
  self::$host = '';
}
```

**Comment by mahone3297** [comment]
@staabm
because 
```
$logger->pushProcessor(new HostProcessor());
```
is more simple than 
```
$logger->pushProcessor(function (array $record) {
    $record['extra']['host'] = $this->host;
    return $record;
});
```

File: src/Monolog/Processor/HostProcessor.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+namespace Monolog\Processor;
+
+use Monolog\Logger;
+
+/**
+ * Injects value of gethostname in all records
+ */
+class HostProcessor
+{
+    private static $host = '';
```
Comment (HonzaMac) [inline src/Monolog/Processor/HostProcessor.php]: Default values is not necessary

File: src/Monolog/Processor/HostProcessor.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+namespace Monolog\Processor;
+
+use Monolog\Logger;
+
+/**
+ * Injects value of gethostname in all records
+ */
+class HostProcessor
```
Comment (HonzaMac) [inline src/Monolog/Processor/HostProcessor.php]: `HostnameProcessor` sounds more aqurate.

Comment (mahone3297) [inline src/Monolog/Processor/HostProcessor.php]: ok, I will change.

File: src/Monolog/Processor/HostProcessor.php (hunk 1 above)
Comment (mahone3297) [inline src/Monolog/Processor/HostProcessor.php]: ok

**Comment by Seldaek** [comment]
Thanks

**Pull request merged**